### PR TITLE
Clean up the smf namespace 

### LINKED ISSUE #4947: Confusion regarding the smf module
This code works fine as expected:

```python
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

n = 100
data = pd.DataFrame({"y": np.random.poisson(lam=1, size=n), 
                     "x": np.random.randn(n)})

result = smf.poisson("y ~ x", data).fit()
result.summary()
```

The problem happens when the user mistakenly tries to use the `Poisson` class (which doesn't use the formula syntax, yet it is imported within the module nonetheless and will even autocomplete if the user makes the mistake of capitalizing the P) within the `smf` class as below,

```python
result = smf.Poisson("y ~ x", data).fit()
result.summary()
```

It is very easy to make this mistake and the resulting error message is quite cryptic, so I am wondering if there is a way to fix this.  It seems that the `smf` module should not directly expose those classes that don't follow the formula syntax.

### PR DESCRIPTION
**Pull request by dsaxton**
Removes unused classes from the smf namespace, closes #4947 and #833.

### CODE REVIEW

File: statsmodels/formula/api.py [hunk 1]
```diff
@@ -1,33 +1,28 @@
...
+probit = dm.Probit.from_formula
+poisson = dm.Poisson.from_formula
+negativebinomial = dm.NegativeBinomial.from_formula
+quantreg = qr.QuantReg.from_formula
+phreg = hr.PHReg.from_formula
+ordinal_gee = gee.OrdinalGEE.from_formula
+nominal_gee = gee.NominalGEE.from_formula
+gee = gee.GEE.from_formula
+
+del lm, dm, mlm, glm, roblm, qr, hr
```
Comment (jbrockmendel) [inline statsmodels/formula/api.py]: Dont delete glm.

**Comment by josef-pkt** [comment]
The naming is too fragile, any mistake leads to many test failures.

I suggest to add a trailing underline, e.g. `glm_` to *all* shortcut module names so they can easily be distinguished from the final_from formula names like `glm`.

Relying on reassigning a different object to the same name will for sure cause problems again in future.

File: statsmodels/formula/api.py [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
-phreg = PHReg.from_formula
-from statsmodels.genmod.generalized_estimating_equations import (GEE,
-     OrdinalGEE, NominalGEE)
-gee = GEE.from_formula
-ordinal_gee = OrdinalGEE.from_formula
-nominal_gee = NominalGEE.from_formula
+import statsmodels.regression.linear_model as lm
+import statsmodels.discrete.discrete_model as dm
+import statsmodels.regression.mixed_linear_model as mlm
+import statsmodels.genmod.generalized_linear_model as glm
```
Comment (bashtage) [inline statsmodels/formula/api.py]: import as genlm not glm and then del genlm below.

**Comment by dsaxton** [comment]
Would something like 

```python
import statsmodels.regression.linear_model as lm_
import statsmodels.discrete.discrete_model as dm_
import statsmodels.regression.mixed_linear_model as mlm_
import statsmodels.genmod.generalized_linear_model as glm_
import statsmodels.robust.robust_linear_model as roblm_
import statsmodels.regression.quantile_regression as qr_
import statsmodels.duration.hazard_regression as hr_
import statsmodels.genmod.generalized_estimating_equations as gee_

gls = lm_.GLS.from_formula
wls = lm_.WLS.from_formula
ols = lm_.OLS.from_formula
glsar = lm_.GLSAR.from_formula
mixedlm = mlm_.MixedLM.from_formula
glm = glm_.GLM.from_formula
rlm = roblm_.RLM.from_formula
mnlogit = dm_.MNLogit.from_formula
logit = dm_.Logit.from_formula
probit = dm_.Probit.from_formula
poisson = dm_.Poisson.from_formula
negativebinomial = dm_.NegativeBinomial.from_formula
quantreg = qr_.QuantReg.from_formula
phreg = hr_.PHReg.from_formula
ordinal_gee = gee_.OrdinalGEE.from_formula
nominal_gee = gee_.NominalGEE.from_formula
gee = gee_.GEE.from_formula

del lm_, dm_, mlm_, glm_, roblm_, qr_, hr_
```

make sense?

**Comment by josef-pkt** [comment]
> make sense?

Yes, I like it because it provides a fast visual check that we only delete trailing underline names and that we don't have any of those left when doing `dir(smf)`

**Comment by jbrockmendel** [comment]
Looks like you nailed it.  Good job

**Comment by josef-pkt** [comment]
all green, merging

@dsaxton Thank you

**Comment by dsaxton** [comment]
@josef-pkt NP, glad to help

**Pull request merged**